src/App.tsx

```diff
@@ -1,26 +1,48 @@
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+import { useEffect, useState } from 'react';
+import { getTodos } from './api';
+import { useDispatch } from 'react-redux';
+import { setTodos } from './features/todos';
+import { useAppSelector } from './app/hooks';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+export const App = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const dispatch = useDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos()
+      .then(res => dispatch(setTodos(res)))
+      .catch(err => {
+        throw err;
```

Comment:
Throwing an error inside the `catch` block without handling it can lead to unhandled promise rejections. Consider handling the error more gracefully, such as by setting an error state and displaying an error message to the user.